src/createStore.js

```diff
@@ -92,8 +93,19 @@ export default function createStore(reducer, initialState) {
       isPlainObject(action),
       'Actions must be plain objects. Use custom middleware for async actions.'
     );
+    invariant(
+      !isDispatching,
+      'Cannot dispatch in the middle of reducer call.'
+    );
+
+    isDispatching = true;
+
+    try {
```

Comment:
We can't add a `try {} catch {}` here. This will cause way more problems then it solves. All reducer errors will be swallowed.